Core/clim-core/frames/define-application-frame.lisp

```diff
@@ -123,7 +129,8 @@
                               ~R).~@:>"
                              key values (length values)))
                     (check-one-value (first values)))
-                   (* (mapcar #'check-one-value values)))))
+                   (*
+                    (check-one-value values)))))
```

Comment:
Is this change correct? mapcar looks sensible here.

Reply:
Both are correct. The `mapcar` version was intended to validate multiple values individually. The new version validates all of `values` at once, which is more powerful and even necessary for example for plist-shaped values. This change does not introduce problems with other options because no option which accepts multiple values had a type specification before this change.

Reply:
The name of the function is inaccurate then.  Either change the name of the function, or introduce a second function:
```
(flet ((check-1-values (values)
                        (destructuring-bind (value &rest rest) values
                          (unless (null rest)
                            (error "~@<The option ~S takes a single ~
                              argument (not ~{~S~^ ~} which are ~
                              ~R).~@:>"
                                   key values (length values)))
                          (when (and type (not (typep value type)))
                            (error "~@<The value ~S for option ~S is ~
                                  not of type ~A.~@:>"
                                   value key type)))
                        value)
                      (check-*-values (values)
                        (when (and type (not (typep value type)))
                          (error "~@<The value ~S for option ~S is ~
                                  not of type ~A.~@:>"
                                 value key type))
                        value))
                 (ecase value-count
                   (1 (check-1-values values))
                   (* (check-*-values values))))
```

I think that the latter solution, while more verbose, is more explicit with putting ephasis, that different things are checked depending on the value-count qualifier.

Reply:
n.b the fact that :type is used differently for different value-count specifiers may warrant a comment too.

Reply:
I commented and refactored the function to make the argument count distinction clearer.